python/hsfs/training_dataset.py

```diff
@@ -194,20 +202,32 @@ def insert(
         # Arguments
             features: Feature data to be materialized.
             overwrite: Whether to overwrite the entire data in the training dataset.
-            write_options: Additional write options as key/value pairs.
-                Defaults to `{}`.
+            write_options: Additional write options as key-value pairs, defaults to `{}`.
+                When using the `hive` engine, write_options can contain the
+                following entries:
+                * key `spark` and value an object of type
+                [hsfs.core.job_configuration.JobConfiguration](../job_configuration)
+                  to configure the Hopsworks Job used to compute the training dataset.
+                * key `wait_for_job` and value `True` or `False` to configure
+                  whether or not to the insert call should return only
+                  after the Hopsworks Job has finished.
 
         # Returns
-            `TrainingDataset`: The updated training dataset metadata object, the
-                previous `TrainingDataset` object on which you call `save` is also
-                updated.
+            `Job`: When using the `hive` engine, it returns the Hopsworks Job
+                that was launched to create the training dataset.
 
         # Raises
             `RestAPIError`: Unable to create training dataset metadata.
         """
-        self._training_dataset_engine.insert(self, features, write_options, overwrite)
+        # td_job is used only if the hive engine is used
+        td_job = self._training_dataset_engine.insert(
+            self, features, write_options, overwrite
+        )
+
+        if engine.get_type() == "spark":
+            self.compute_statistics()
```

Comment:
where does this come from?

Reply:
I thought that it was a bug and the statistics computation was triggered with the `hive` engine. Actually the `compute_statistics()` contains already a check for the engine type.